packages/router/src/history.tsx

```diff
@@ -15,20 +15,10 @@ const createHistory = () => {
       return listenerId
     },
     navigate: (to: string, options?: NavigateOptions) => {
-      const { pathname, search, hash } = new URL(
-        globalThis?.location?.origin + to
-      )
-
-      if (
-        globalThis?.location?.pathname !== pathname ||
-        globalThis?.location?.search !== search ||
-        globalThis?.location?.hash !== hash
-      ) {
-        if (options?.replace) {
-          globalThis.history.replaceState({}, '', to)
-        } else {
-          globalThis.history.pushState({}, '', to)
-        }
```

Comment:
The same location check is obsolete with the patch of globalThis.history.replaceState and globalThis.history.pushState